src/ImageSharp/Formats/Tiff/ITiffEncoderOptions.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.Compression.Zlib;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.Processing.Processors.Quantization;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    /// <summary>
+    /// Encapsulates the options for the <see cref="TiffEncoder"/>.
+    /// </summary>
+    internal interface ITiffEncoderOptions
+    {
+        /// <summary>
+        /// Gets the number of bits per pixel.
+        /// </summary>
+        TiffBitsPerPixel? BitsPerPixel { get; }
+
+        /// <summary>
+        /// Gets the compression type to use.
+        /// </summary>
+        TiffCompression Compression { get; }
+
+        /// <summary>
+        /// Gets the compression level 1-9 for the deflate compression mode.
+        /// <remarks>Defaults to <see cref="DeflateCompressionLevel.DefaultCompression"/>.</remarks>
+        /// </summary>
+        DeflateCompressionLevel CompressionLevel { get; }
+
+        /// <summary>
+        /// Gets the encoding mode to use. Possible options are RGB, RGB with a color palette, gray or BiColor.
+        /// If no mode is specified in the options, RGB will be used.
+        /// </summary>
+        TiffEncodingMode Mode { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether to use horizontal prediction. This can improve the compression ratio with deflate or lzw compression.
+        /// </summary>
+        bool UseHorizontalPredictor { get; }
```

Comment:
Is the horizontal predictor the only filter option?

Reply:
Yes, its the only one.

Reply:
Looks like there's 3 options in `TiffPredictor`

Reply:
Ah yes, i forgot about this. This is not part of the Tiff Spec 6.0 as far as i know. I think it was introduced later. It works the same as the current horizontal predictor, but is meant for float pixel types, which we do not support yet. 

Should we keep this as a bool, or introduce an enum?

Reply:
ok changed with de6e9ddb